Training24/Test.c

```diff
@@ -1,19 +1,150 @@
 // ------------------------------------------------------------------------------------
 // Training ~ A training program for new joiners at Metamation, Batch - July 2024.
 // Copyright (c) Metamation India.
-// -------------------------------------
-//  Dhaarani Mohanasundaram
-// -------------------------------------
-// Standard Test file in Main
+// Dhaarani Mohanasundaram
 // -------------------------------------
 // Test.c
-// Program on main branch.
+// Program on A5 branch.
+// Program that implements test cases and checks the user input for sort & search.
 // ------------------------------------------------------------------------------------
 #include <stdio.h>
+#include <stdbool.h>
 #include "Header.h"
 
-int main() {
-const char* message = GetMessage ();
-printf ("%s\n", message);
-return;
+#define RED "\x1b[31m"
+#define GREEN "\x1b[32m"
+#define CYAN "\x1b[36m"
+#define RESET "\x1b[0m"
+#define MAX_SIZE 100
+
+/// <summary>Function to clear the newline characters.</summary>
+void ClearInputBuffer ();
+
+/// <summary>Function to print the test table row.</summary>
+void PrintTestTableRow (const int* unsortedArr, int n, const int* sortedArr, const int* expectedSortedArr, int searchElement, int actualIndex, int expectedIndex);
+
+int main () {
+   int choice;
+
+   while (1) {
+      printf ("\nChoose an option:\n" "1. Run Test Cases\n" "2. Enter User Input\n" "3. Exit\n" "\nEnter your choice (1-3): ");
+      if (scanf ("%d", &choice) != 1) {
+         printf ("Invalid input. Please enter a number between 1 and 3.\n");
+         ClearInputBuffer ();
+         continue;
+      }
+      switch (choice) {
+      case 1:
+         ExecuteTestCases ();
+         break;
+      case 2:
+         ExecuteUserInput ();
+         break;
+      case 3:
+         printf ("\nExiting the program...!\n");
+         return 0;
+      default:
+         printf ("Invalid choice. Please enter a number between 1 and 3.\n");
+         break;
+      }
+   }
+}
+
+
+void ExecuteTestCases () {
+   int testCases[][MAX_SIZE] = { {1, 2, 3, 4, 5, 6, 7, 8}, {0, -3, 1, 3, 5, 7, -9}, {10, -20, -30, 40, 0}, {3, -3, 3, -3, -3, -3} };
+   int expectedSorted[][MAX_SIZE] = { {1, 2, 3, 4, 5, 6, 7, 8}, {-9, -3, 0, 1, 3, 5, 7}, {-30, -20, 0, 10, 40}, {-3, -3, -3, -3, 3, 3} };
+   int expectedIndex[][2] = { {4, 3}, {5, 5}, {20, -1}, {3, 4} }; // -1 is element not found
+   int sizes[] = { 8, 7, 5, 6 }; // Sizes of each test case
```

Comment:
Since all are int put in a single int declaration. Apply this wherever possible.
```suggestion
   int testCases[][MAX_SIZE] = { {1, 2, 3, 4, 5, 6, 7, 8}, {0, -3, 1, 3, 5, 7, -9}, {10, -20, -30, 40, 0}, {3, -3, 3, -3, -3, -3} }, 
   expectedSorted[][MAX_SIZE] = { {1, 2, 3, 4, 5, 6, 7, 8}, {-9, -3, 0, 1, 3, 5, 7}, {-30, -20, 0, 10, 40}, {-3, -3, -3, -3, 3, 3} }, 
   expectedIndex[][2] = { {4, 3}, {5, 5}, {20, -1}, {3, 4} }, 
   sizes[] = { 8, 7, 5, 6 }; // Sizes of each test case
```